pandas/core/frame.py

```diff
@@ -49,10 +49,17 @@
 )
 from pandas._libs.hashtable import duplicated
 from pandas._libs.lib import is_range_indexer
-from pandas.compat import PYPY
+from pandas.compat import (
+    PYPY,
+    pa_version_under11p0,
+)
 from pandas.compat._constants import REF_COUNT
 from pandas.compat._optional import import_optional_dependency
 from pandas.compat.numpy import function as nv
+
+if not pa_version_under11p0:
```

Comment:
What prevents this from working with pyarrow 10?

Reply:
Nothing prevents it ! :-) 

It was a misanderstanding. 

At the beginning I thought that the version 11.0.0 was the minimum version (because "Minimum Versions" tests in the CI did not work). But actually `value` in `_sanitize_column` is:
- A `pa.lib.ChunkedArray` in version `10.0.1` 
- A `pa.lib.Array` in versions above
So now I just test if `value` is an instance of the parent class `pa.lib._PandasConvertible`